Cargo.toml

```diff
@@ -12,22 +12,22 @@ categories = ["science"]
 keywords = ["GRIB", "weather", "meteorology"]
 
 [dependencies]
-chrono = "0.4"
-clap = "2.33.0"
-console = "0.11"
-num = "0.3.1"
-num_enum = "0.5.1"
+chrono = "0.4.19"
```

Comment:
For a library crate these bounds are usually a bit too strict. Applications (i.e. whoever uses this grib-rs) have a `Cargo.lock` file that will resolve the exact crate versions. The reason that library crates should NOT specify dependencies too strictly is that patches and security fixes would take forever to trickle down the dependency chain. The Rust ecosystem uses [Semantic Versioning](https://semver.org/) which ensures that some assumptions can be made. So my recommendations are:

- `chrono = "0.4"` actually resolves to `chrono >= 0.4 AND chrono < 0.5`, so the existing bounds are OK
- Same goes for many other crates. For `0.x` crates is usually sufficient to pin `0.x`. For stable crates I would go for just the major version (e.g `which = "4"`).
- If you need at least a specific feature or patch release, you can use something like `clap = "^2.33"` (resolves to `clap > 2.33 AND clap < 3`). You might wanna have a look at the [Cargo documentation](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies).

Reply:
@crepererum 
Thank you very much for reviewing. I had a similar impression and was about to make a similar comment about the preferable width of the version specification for the library dependencies and version strategies against unstable/stable API dependencies.

Since I agree with the main idea of this PR from @Quba1, I would like to accept it and then revert some dependencies that are too strict. I will also make sure to document those policies on dependencies.

Reply:
(This is just a follow-up comment.)

Also, in general, there is a problem that too strict dependencies can lead to loss of interoperability and dependency hell.  However, in Rust, `libA -> libB [v1]` and `libA -> libC -> libB [v2]` can coexist without dependency conflicts, so there is no problem even if the dependencies are strict.